main/lsp/LSPConfiguration.cc

```diff
@@ -213,6 +213,14 @@ unique_ptr<Location> LSPConfiguration::loc2Location(const core::GlobalState &gs,
     return make_unique<Location>(uri, std::move(range));
 }
 
+vector<string> LSPConfiguration::frefsToPaths(const core::GlobalState &gs, const vector<core::FileRef> &refs) const {
+    vector<string> paths;
+    paths.resize(refs.size());
+    std::transform(refs.begin(), refs.end(), paths.begin(),
```

Comment:
I think `std::transform` is OK to use. I use `for` loops in many places where I may:

* Need the iterator to remove something while I iterate (used in preprocessor's mergeFileContents method)
* Need to break out of the loop
* Need the iterator to begin an inner loop (used in mergeFileContents)
* Need an excessive amount of captured state in the lambda 

I might have pushed back on using transform elsewhere in the cases above. (Otherwise, mea culpa!)

I don't remember when I wrote this code, but I think I used transform here because it satisfies the following constraints:

* It doesn't resize the lambda while iterating over the input vector (we `resize` it up front), unlike a `push_back` solution that may periodically need to allocate and copy the vector. Maybe I thought this would be called with a large set of files; looking at callsites, it's only called on the vector of files with errors in production.
* It's compact and straightforward, unlike a loop-based solution that would need to manually track indexes (e.g. `output[i] = string(input[i])`).
* The transformation is a pure function that only needs to (const) capture global state.